### PR TITLE
Split libPlatform

### PR DESCRIPTION
**Pull request by gmrukwa**
This is the subset of changes I am rushing in the run of molecular signature estimation framework. I thought it'll be useful for you to see it as a self-contained subpart.

Here I've splitted `libPlatform` to separate the functionalities gathered there. This way, we'll not push too much responsibilities into this project.

Right now we will have two projects:

* `libFunctional` - here are all functional-alike extensions for C++ we're developing
* `libStatistics` - here I will push some statistical indexes, etc., just to gather them together

### CODE REVIEW

**Comment by gmrukwa** [comment]
[Refactor libPlatform](https://trello.com/c/m2PRlhce/309-refactor-libplatform)

File: src/Spectre.libClassifier.Tests/Spectre.libClassifier.Tests.vcxproj.filters [hunk 1]
```diff
@@ -27,10 +27,10 @@
     <ClCompile Include="RandomSplitterTest.cpp">
       <Filter>Source Files</Filter>
     </ClCompile>
-    <ClCompile Include="C:\Users\regis\Documents\GitHub\spectre\src\Spectre.libClassifier.Tests\..\..\lib\native\src\gtest\gtest-all.cc">
+    <ClCompile Include="C:\Users\Grzegorz\Source\Repos\MSI-dev\spectre\src\Spectre.libClassifier.Tests\..\..\lib\native\src\gtest\gtest-all.cc">
```
Comment (Sand3r-) [inline src/Spectre.libClassifier.Tests/Spectre.libClassifier.Tests.vcxproj.filters]: Nice one

File: src/Spectre.libClassifier.Tests/Spectre.libClassifier.Tests.vcxproj.filters [hunk 2]
```diff
@@ -27,10 +27,10 @@
...
       <Filter>Source Files</Filter>
     </ClCompile>
-    <ClCompile Include="C:\Users\regis\Documents\GitHub\spectre\src\Spectre.libClassifier.Tests\..\..\lib\native\src\gmock\gmock-all.cc">
+    <ClCompile Include="C:\Users\Grzegorz\Source\Repos\MSI-dev\spectre\src\Spectre.libClassifier.Tests\..\..\lib\native\src\gmock\gmock-all.cc">
```
Comment (Sand3r-) [inline src/Spectre.libClassifier.Tests/Spectre.libClassifier.Tests.vcxproj.filters]: ^

File: src/Spectre.libFunctional.Tests/Spectre.libFunctional.Tests.vcxproj.filters [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    <ClCompile Include="FindTest.cpp">
+      <Filter>Source Files</Filter>
+    </ClCompile>
+    <ClCompile Include="FilterTest.cpp">
+      <Filter>Source Files</Filter>
+    </ClCompile>
+    <ClCompile Include="TransformTest.cpp">
+      <Filter>Source Files</Filter>
+    </ClCompile>
+    <ClCompile Include="C:\Users\Grzegorz\Source\Repos\MSI-dev\spectre\src\Spectre.libFunctional.Tests\..\..\lib\native\src\gtest\gtest-all.cc">
```
Comment (Sand3r-) [inline src/Spectre.libFunctional.Tests/Spectre.libFunctional.Tests.vcxproj.filters]: And also in these 2:

File: src/Spectre.libFunctional.Tests/Spectre.libFunctional.Tests.vcxproj.filters [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      <Filter>Source Files</Filter>
+    </ClCompile>
+    <ClCompile Include="C:\Users\Grzegorz\Source\Repos\MSI-dev\spectre\src\Spectre.libFunctional.Tests\..\..\lib\native\src\gmock\gmock-all.cc">
```
Comment (Sand3r-) [inline src/Spectre.libFunctional.Tests/Spectre.libFunctional.Tests.vcxproj.filters]: ^

**Pull request merged**